AsyncDisplayKit/ASDisplayNode+Subclasses.h

```diff
@@ -155,6 +156,19 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)invalidateCalculatedLayout;
 
 
+/** @name Layout Transitioning */
+
+/**
+ @discussion A place to perform your animation. New nodes have been inserted here. You can also use this time to re-order the hierarchy.
```

Comment:
Great place to have developers encode any specific ordering requirements, if any are needed - this combined with being able to use addSubnode in init (particularly for nodes that don't need transitions) means that we should feel confident without additional handling of the hierarchy "z-order" management.